draft/ZEP0002.md

```diff
@@ -0,0 +1,285 @@
+
+Sharding could be implemented directly in the storage layer, but to allow interoperability between different stores and Zarr implementations, it is important to be able to consistently adress array data within a store. Therefore, standardizing the sharding format across different stores is crucial for interoperability, which is possible by implementing sharding as a storage transformer.
+
+
+## Usage and Impact
+
+Sharding allows to store very large, chunked arrays, which currently would have inefficient access characteristics or might even be impossible to store as generic Zarr arrays.
+
+Image datasets in the biomedical, geo-science and general data-science communities are steadily growing. In recent years, peta-scale datasets have been made publicly available. Although peta-scale datasets are still infrequent, tera-scale datasets are acquired routinely and rapidly grow in size further.
+
+Currently, these large-scale datasets are made available through tool-specific formats, such as neuroglancer-precomputed or webknossos-wrap, which support sharding. While the data is accessible and explorable in individual software tools, wider interoperability with other software tools is limited. Whereas it has been possible to convert smaller datasets to work in different tools, it becomes infeasible for large-scale datasets. The choice of tool-specific formats is in part a result of the lack of a standard format that makes storage and processing of very large datasets practical. Providing sharding in Zarr would allow Zarr to become the standard format also for tera- and peta-scale array data and future-proofing it as a standard for the requirements of growing data sizes in the scientific community.
+
+
+## Backward Compatibility
+
+This ZEP introduces a Zarr extension for Zarr version 3. The sharding
+extension is optional for data producers, and only necessary when reading
+arrays that are written using the sharding extension. It corresponds
+to a minor version change in semantic versioning, adding functionality
+in a backwards compatible manner.
+
+
+## Detailed description
+
+
+### Combining regular blocks of chunks in a shard
+
+To add sharding to Zarr, data that was stored previously as separate objects must be combined.
+Different strategies can be considered when sharding array data, such as:
+1. sharding across single dimensions,
+2. sharding by regular n-dimensional blocks of data, and
+3. sharding random combinations of chunks via hashing.
+
+All of those strategies only shard the array-data, both as metadata must first be read without
+sharding, so that sharding itself can be configured in the metadata, and also since the relevant
+parts of the data is the array data itself.
```

Comment:
might be a matter of taste, but I think this can be put a bit more clearly. Something like this:

```suggestion
All of those strategies only shard the array-data and not the metadata, since the latter is not big enough to warrant sharding and also because the sharding configuration is stored in the metadata.
```